### ISSUE TITLE
[question] integrity verification of local packages

### ISSUE THREAD

**Issue by dzolee** [conan-io/conan#4019]
Hello there,

I have experienced a situation where half of the package files (headers, libraries) just disappeared from the client computer. The reasons are irrelevelant here, might be a bored user or a technical error in the storage. Bottom line: Conan (1.9.1) does not detect that the package is broken, my project is configured and after 30 minutes of build time, I get errors because libs or headers are missing. I have looked into the docs but have not found a way to force a "sanity check" on the local packages. Is there such a way?

**Comment by jgsogo** [comment]
Hi @dzolee!

We assume that the user is not interacting with the files in the cache, the cache should be modified only through Conan commands; so there can be some fails associated with this unexpected environment.

At this moment you can implement a [_hook_](https://docs.conan.io/en/latest/reference/hooks.html#hooks-reference) for the `pre_build` action in the consuming package, here you can traverse all the dependencies checking whatever you need:

```python
def pre_build(output, conanfile, **kwargs):
    for dep in conanfile.deps_cpp_info.deps:
        dep_values = conanfile.deps_cpp_info[dep]
        print(dep_values.rootpath)
        
```

Let me know if it will solve your issue or maybe you have a better idea on how to solve it.

Thanks!

**Comment by dzolee** [comment]
Hi there,

I see the conanmanifest.txt for TBB and Boost (installed from conan/stable) and it contains a pretty complete list of files with some kind of fingerprint:

```
1541879075
bin/tbb.dll: 67c01fe288755bd568a04ce200c7915e
bin/tbbmalloc.dll: 2ed71a1df986d85b107a01728836350c
bin/tbbmalloc_proxy.dll: 847909578442f4799561d1197af12b1a
conaninfo.txt: ed801f2e38c4a0a40c315d47f75de35b
include/serial/tbb/parallel_for.h: cbc1275e9a34dd943d4d1d31863b15bf
include/serial/tbb/tbb_annotate.h: cb129f0cecff46bdea633d23db82a17f
include/tbb/aggregator.h: ec1e5d9da4fff99945f2d8682245231f
include/tbb/aligned_space.h: 14b5ebf883388965cc48a4569dab25f2
include/tbb/atomic.h: ffee72067a854990c8b4d7543bac9e77
include/tbb/blocked_range.h: db01ac956be31442f38511956f56cb07

```
can we somehow trigger a local package integrity verification using the above infos? The list of files that need to be in the package and their hash seems to be recorded ...?

**Comment by memsharded** [comment]
Yes, there is already a opened issue for this, likely that it will be implemented. 
Please check it here: https://github.com/conan-io/conan/issues/3821 and if it is what you are looking for, please close this issue.

**Comment by memsharded** [comment]
Totally, it will be opt-in. It was there in place, as default, and we had to remove it because it was very slow.
